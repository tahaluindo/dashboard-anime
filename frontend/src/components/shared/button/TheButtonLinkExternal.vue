<template>
  <a :href="href">
    <slot />
  </a>
</template>

<script lang="ts">
import { defineComponent } from "vue";

export default defineComponent({
  props: {
    href: {
      type: String,
      required: true,
    },
  },
});
</script>