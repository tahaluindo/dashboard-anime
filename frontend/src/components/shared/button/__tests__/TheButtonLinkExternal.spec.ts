import { mount } from "@vue/test-utils";
import TheButtonLinkExternal from "../TheButtonLinkExternal.vue";

describe("TheButtonLinkExternal component test", () => {
  it("should render the default component", () => {
    const href = "https://test.com";
    const wrapper = mount(TheButtonLinkExternal, {
      props: {
        href,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
