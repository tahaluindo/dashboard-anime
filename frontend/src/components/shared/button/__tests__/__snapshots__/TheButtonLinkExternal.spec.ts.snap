// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheButtonLinkExternal component test should render the default component 1`] = `"<a href=\\"https://test.com\\"></a>"`;
